actors/builtin/shared.go

```diff
@@ -54,3 +56,27 @@ type MinerAddrs struct {
 type ConfirmSectorProofsParams struct {
 	Sectors []abi.SectorNumber
 }
+
+// ResolveToIDAddr resolves the given address to it's ID address form.
+// If an ID address for the given address dosen't exist yet, it tries to create one by sending a zero balance to the given address.
+func ResolveToIDAddr(rt runtime.Runtime, address addr.Address) (addr.Address, error) {
+	// if we are able to resolve it to an ID address, return the resolved address
+	idAddr, found := rt.ResolveAddress(address)
+	if found {
+		return idAddr, nil
+	}
+
+	// send 0 balance to the account so an ID address for it is created and then try to resolve
+	_, code := rt.Send(address, MethodSend, nil, abi.NewTokenAmount(0))
+	if !code.IsSuccess() {
+		return address, fmt.Errorf("failed to send zero balance to account %v, got code %v", address, code)
```

Comment:
`code` here is an `ExitCode` which implements error, so we should return a thing that will result in propagating that exit code. I think you want something like `code.Wrapf(msg, args)`.